src/slic3r/GUI/PhysicalPrinterDialog.cpp

```diff
@@ -508,19 +508,21 @@ void PhysicalPrinterDialog::update(bool printer_change)
             m_optgroup->show_field("printhost_apikey", auth_type == AuthorizationType::atKeyPassword);
             for (const char* opt_key : { "printhost_user", "printhost_password" })
                 m_optgroup->show_field(opt_key, auth_type == AuthorizationType::atUserPassword);
+        // hide api key and CA file for MPMDv2
+        } else if (opt && opt->value == htMPMDv2) {
+            m_optgroup->hide_field("printhost_apikey");
+            m_optgroup->hide_field("printhost_cafile");
+        // hide api key for klipper
+        } else if (opt && opt->value == htKlipper) {
+            m_optgroup->hide_field("printhost_apikey");
         } else {
             m_optgroup->hide_field("printhost_authorization_type");
             m_optgroup->show_field("printhost_apikey", true);
+            m_optgroup->show_field("printhost_cafile", true);
```

Comment:
Otherwise the cafile field will not show up once you select one of the other options that uses it from the dropdown.

Or in other words, if you switch between the options in the dropdown we can't just hide fields, we need to show them again as required.

This is pretty ugly from the start imho. Would probably need some work to make it more modular. Possibly a separate dialog for each print host option or something.